### ISSUE TITLE
Dashed lines do not appear as dashed in the legend.

### ISSUE THREAD

**Issue by mathiasleroy** [apache/echarts#9552]
It would be nice if the legend symbols adapted to the series symbols.  
In my case, dashed lines do not appear as dashed in the legend.

![influenza-like illness_ incidence of gp consultations by season](https://user-images.githubusercontent.com/2870437/49641921-e4835380-fa11-11e8-8b06-676c1c75f6c7.png)

By default we have this big circle in the middle of the line, I got rid of it with 

``` 
symbol: 'line', symbolSize:0,
```

_____

- [x] I am using English in this issue. 在这个 Issue 中我使用了英文（强烈建议）。
- [x] Required: I have read the document and examples and tried to solve it by myself. （必填）我读过了文档和教程，并且曾试图自己解决问题。
- [x] Required: I have searched for similar issues and that didn't help. （必填）我搜索过 issue 但是没有帮助。
- [x] Required: I have tried with the latest version of ECharts and still have this problem. （必填）我试过最新版本的 ECharts，还是存在这个问题。

**Comment by Ovilia** [comment]
@mathiasleroy Yes, that would be a great feature. In fact, we'd like to improve legend style not only with dashed lines, but also other features like line width, stroke styles.

**Comment by Julien76** [comment]
Hi,
Any update on this feature?
I found it quite hard to style the legend properly.

**Comment by kakusx** [comment]
Same problem.

**Comment by JerryLeeCS** [comment]
Any update on this?

**Comment by natalidedik** [comment]
Will it be fixes? That's really look like a shortcoming that have to be fixed!!!

**Comment by cvetkovivan** [comment]
So, i found one solution. 
`EchartOptions.legend.data[] = {name: '', icon: ''}`

You can add any icon to current data in your legend.
Exapmle link: [codepen](https://codepen.io/usfbh/pen/BayMbzX).
Also you can add svg icon. [documantation](https://echarts.apache.org/en/option.html#legend.data.icon).

No idea to resovle this problem by any other solution.

**Comment by JJZolper** [comment]
Yes any update?

**Comment by JJZolper** [comment]
My client is asking for this and besides trying to get wild with an image it'd be nice if we could get movement on a real solution.

**Comment by mathiasleroy** [comment]
please don't close this issue

**Comment by Civilduino** [comment]
this feature is needed and additional flexibility with the legend symbol format, right now is very limited

**Comment by sovushka-utrom** [comment]
@mathiasleroy,
> symbol: 'line', symbolSize:0,   

Thank you for this one!

As for the topic, adding
```
    lineStyle: {
      type: 'dashed',
      width: 1.5, // setting width is important to have regular dashes 
    },
```
to the legend does the trick along with the line above in `series`.

**Comment by mlaffargue** [comment]
Even if it should be the default value as stated in the  ([documentation](https://echarts.apache.org/en/option.html#legend.itemStyle.borderType)), forcing the legend itemStyle to inherit will make dashes appear.
```
    legend: {
        show: true,
        data: ['serie1'],
        itemStyle: {
            borderType: 'inherit' // Should be the default value
        }
    },...


    series: [{
        name: "serie1",
        itemStyle: {
            borderType: 'dashed'
        },...
```
